spec/syntax.md

```diff
@@ -87,285 +87,253 @@ The syntax specification takes into account the following design restrictions:
-_declarations_, _selectors_, _functions_, and more.
+Embedding a _pattern_ in brackets ensures that simple _messages_ can be embedded into
+various formats regardless of the container's whitespace trimming rules.
 
-A complex message with 2 _selectors_ and 3 local variable _declarations_:
+> Example. In a Java `.properties` file, the message `hello` has exactly three spaces before and after
+> the word "Hello":
+> ```properties
+> hello = {  Hello  }
+> ```
 
-    let $hostName = {$host :person firstName=long}
-    let $guestName = {$guest :person firstName=long}
-    let $guestsOther = {$guestCount :number offset=1}
+### Text
 
-    match {$host :gender} {$guestOther :number}
+**_text_** is the translatable content of a _pattern_.
+Any Unicode code point is allowed,
+except for surrogate code points U+D800 through U+DFFF.
+The characters `\`, `{`, and `}` MUST be escaped as `\\`, `\{`, and `\}`.
 
-    when female 0 {{$hostName} does not give a party.}
-    when female 1 {{$hostName} invites {$guestName} to her party.}
-    when female 2 {{$hostName} invites {$guestName} and one other person to her party.}
-    when female * {{$hostName} invites {$guestName} and {$guestsOther} other people to her party.}
+All code points are preserved. Whitespace in text is significant.
 
-    when male 0 {{$hostName} does not give a party.}
-    when male 1 {{$hostName} invites {$guestName} to his party.}
-    when male 2 {{$hostName} invites {$guestName} and one other person to his party.}
-    when male * {{$hostName} invites {$guestName} and {$guestsOther} other people to his party.}
+```abnf
+text = 1*(text-char / text-escape)
+text-char = %x0-5B         ; omit \
+          / %x5D-7A        ; omit {
+          / %x7C           ; omit }
+          / %x7E-D7FF      ; omit surrogates
+          / %xE000-10FFFF
+```
 
-    when * 0 {{$hostName} does not give a party.}
-    when * 1 {{$hostName} invites {$guestName} to their party.}
-    when * 2 {{$hostName} invites {$guestName} and one other person to their party.}
-    when * * {{$hostName} invites {$guestName} and {$guestsOther} other people to their party.}
+### Placeholder
 
-## Productions
+A **_placeholder_** is another word for an _expression_ that appears inside of a _pattern_
+and which will be replaced during the formatting of the _message_.
 
-The specification defines the following grammar productions.
+### Selector
 
-A message satisfying all rules of the grammar is considered _well-formed_.
+A **_selector_** selects a specific _pattern_ from a list of available
+_variants_ in a _message_.
+_Selectors_ provide the ability for a _message_ to use a _pattern_ that 
+varies in content or form depending on values determined at runtime.
 
-Furthermore, a well-formed message is considered _valid_
-if it meets additional semantic requirements about its structure, defined below.
+A _selector_ consists of a _match_ statement followed by at least one _variant_.
```

Comment:
I've always thought about this the other way around: expressions can either be placeholders or selectors, and the `match` statement is where you put multiple selectors into.

Also, I'd consider variants to be part of the `match` statement, rather than follow after it.

So in my mind it's (pseudo-ABNF):

```abnf
match-stmt = 'match' 1*selector 1*variant
```

rather than:

```abnf
selector = 'match' 1*expression 1*variant
```

Reply:
I like your reformulation very much. It solves a lot of circular jargon problems surrounding what the `expression`s in `selectors` are.